### DISCUSSION TITLE
2FA OK, but not any code saving! Don't ask for it!

### DISCUSSION THREAD

**Discussion by arnvidhr** [community/community#77946] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

I would prefer to use E-mail 2FA, just like such option available everywhere, as more reliable. As you can loose the device with an app, eg. cell phone after what access to your account can be lost, via inability to use app or to receive SMS messages. And e-mail 2FA is easier to implement than an app or SMS 2FA. 

However, since this is too hard for you, so simple feature, I will be forced to use what you have. That would be ok, except one key exception. I don't want to use codes? Please no! Now you enforce mandatory that all accepted to save codes. What the hell is this? I never in my life saved any codes! I don't consider such option as right or valid! While you don't give any other choice! This is unacceptable! 

You even asking to confirm formally that I SAVED DAMN YOUR CODES! But I don't want to save them and this is matter of principle! Where is variety of possible approaches to use the software? Why only one way is allowed? Whose idea that was? Is he or she already fired out, I would ask?

Fix this that codes saved only these who wish to save them, not everybody. This is unacceptable nonsense. I already tried both 2FA ways and both asking confirmation that I saved stupid codes. I won't click it ever. I won't confirm what I not planning to do. Let us to skip it and just do regular 2FA without damn codes saving.

P.S. And fire out that guy, which idea that was for your own sake! If won't be possible to use github without it, I probably will abandon it. This is not discussion. Just wanted to say the opinion which is firm.

**Comment by VIPLightning** [comment]
This is Microsoft's doing and 2FA adds potentially greater security risk. That's pretty much what's happened to 100 million epic games users that were hacked by the Russians. But I digress. I'd switch and migrate all my work to [Codeberg](https://codeberg.org/) or other git websites before December 15 at 0:00 (UTC) arrives and shit hits the fan. Because when it happens, Microsoft pretty much made this website a high valuable target to hit and Microsoft's gonna do what's only in their best interest and not for consumers or developers.

**Comment by arnvidhr** [comment]
M$ Often does it. Their recovery questions with every version of windows - brutal security hole. Just guess stupid words and you are in, even without password! Their added various pass keys, codes, just brute force with fun any of them, which is easier than password or other cases of vulnerabilities, without thinking that even passwordless systems are possible, not only without recovery questions or  other vulnerabilities.

They suppose that everyone using password managers, where they even suggesting to keep codes, not considering that solutions without any codes are possible. 

Like I always telling about Microsoft - they are trying to code, when they don't know how to code, just like me.
